File: external-libs/core/src/main/java/com/serhii/core/security/impl/hash/HashAlgorithms.java
package com.serhii.core.security.impl.hash;

import com.serhii.core.CoreEngine;

/**
 *  Class implements specific hash algorithms
 *
 *  Uses underling OpenSSL implementation
 */
public class HashAlgorithms implements HashGenerator {

    @Override
    public String makeHashMD5(String message) {
        return hashMD5(message);
    }

    private native String hashMD5(String message);

    static {
        System.loadLibrary(CoreEngine.RUNTIME_LIBRARY);
    }
}

File: external-libs/android-core/src/main/java/com/serhii/core/security/impl/hash/HashAlgorithms.kt
/*
 * Copyright 2022. Happy coding ! :)
 * Author: Serhii Butryk
 */
package com.serhii.core.security.impl.hash

import com.serhii.core.CoreEngine.loadNativeLibrary
import com.serhii.core.security.impl.hash.HashGenerator
import com.serhii.core.CoreEngine

/**
 * Class implements specific hash algorithms
 * Uses underling OpenSSL implementation
 */
internal class HashAlgorithms : HashGenerator {

    override fun makeHashMD5(message: String): String {
        return hashMD5(message)
    }

    private external fun hashMD5(message: String): String

    companion object {
        init {
            loadNativeLibrary()
        }
    }

}